### PR TITLE
Skip checking and redispatch in reachability when seeing duplicates

### PR DESCRIPTION
**Pull request by josephschorr**
Note that this just skips within the current dispatch, not globally (that should be handled by cache)

### CODE REVIEW

File: internal/graph/reachableresources.go [hunk 1]
```diff
@@ -36,11 +37,22 @@ type ValidatedReachableResourcesRequest struct {
 	Revision decimal.Decimal
 }
 
+type syncONRSet struct {
+	items sync.Map
+}
+
+func (s *syncONRSet) Add(onr *core.ObjectAndRelation) bool {
+	key := tuple.StringONR(onr)
+	_, existed := s.items.LoadOrStore(key, true)
```
Comment (jzelinskie) [inline internal/graph/reachableresources.go:46]: I think the empty struct instead of `true` trick to save memory should work for sync.Map, too.

Comment (josephschorr) [reply internal/graph/reachableresources.go:46]: Looks like `nil` works too... maybe I should just use that?

Comment (jzelinskie) [reply internal/graph/reachableresources.go:46]: if the implementation of sync.Map uses a `map[any]any`, the go compiler will optimize `struct{}{}` to consume 0 bytes vs nil which is probably going to be sizeof(pointer)

Comment (josephschorr) [reply internal/graph/reachableresources.go]: Updated

**Pull request merged**